core/peerset/src/lib.rs

```diff
@@ -47,8 +40,84 @@ struct Inner {
 	in_slots: Vec<Option<PeerId>>,
 }
 
-/// Message that can be sent by the peer set manager (PSM).
 #[derive(Debug)]
+enum Action {
```

Comment:
Sounds like a good idea for proc-macro library :)

Reply:
definitely :) if we only have more messages or if we do it in more than one place. Right now proc-macro library would probably require writing more code than this boilerplate does